crates/circuit/src/register.rs

```diff
@@ -0,0 +1,1054 @@
+                    bits.get_index_of(bit_inner)
+                        .ok_or(PyValueError::new_err(format!(
+                            "Bit {} not found in Register {}.",
+                            bit,
+                            slf.repr()?,
+                        )))
+                }
+            },
+        }
+    }
+
+    /// Make object hashable, based on the name and size to hash.
+    fn __hash__(slf: Bound<'_, Self>) -> PyResult<isize> {
+        let borrowed = slf.borrow();
+        (slf.get_type(), borrowed.name(), borrowed.size())
+            .into_bound_py_any(slf.py())?
+            .hash()
+    }
+
+    /// Two Registers are the same if they are of the same type
+    /// (i.e. quantum/classical), and have the same name and size. Additionally,
+    /// if either Register contains new-style bits, the bits in both registers
+    /// will be checked for pairwise equality. If two registers are equal,
+    /// they will have behave identically when specified as circuit args.
+    ///
+    /// Args:
+    ///     other (Register): other Register
+    ///
+    /// Returns:
+    ///     bool: `self` and `other` are equal.
+    fn __eq__(slf: Bound<'_, Self>, other: Bound<'_, Self>) -> PyResult<bool> {
+        if slf.is(&other) {
+            return Ok(true);
+        }
+        let slf_borrow = slf.borrow();
+        let other_borrow = other.borrow();
+        Ok(slf.get_type().eq(other.get_type())?
+            && slf.repr()?.to_string().eq(&other.repr()?.to_string())
+            && slf_borrow.eq(&other_borrow))
+    }
+
+    fn __reduce__(slf: Bound<'_, Self>) -> PyResult<Bound<PyTuple>> {
+        let borrowed = slf.borrow();
+        let ty = slf.get_type();
+        let args = match borrowed.0.as_ref() {
+            RegisterInfo::Owning(reg) => (Some(reg.size), Some(reg.name.clone()), None),
+            RegisterInfo::Alias { name, .. } => (
+                None,
+                Some(name.to_string()),
+                Some(PyList::type_object(slf.py()).call1((&slf,))?),
+            ),
+        };
+        (ty, args).into_pyobject(slf.py())
+    }
+
+    fn __iter__(slf: PyRef<'_, Self>) -> PyResult<Bound<'_, pyo3::types::PyIterator>> {
+        return PyList::new(slf.py(), slf.iter().map(PyBit))?.try_iter();
+    }
+
+    /// Return register size.
+    fn __len__(slf: PyRef<'_, Self>) -> usize {
+        slf.size()
+    }
+
+    /// Prefix to use for auto naming.
+    #[classattr]
+    fn prefix<'py>() -> &'py str {
+        "reg"
+    }
+
+    /// Bit type stored in the register.
+    #[classattr]
+    fn bit_type(py: Python) -> Bound<PyType> {
+        PyBit::type_object(py)
+    }
+
+    /// Counter for the number of instances in this class.
+    #[classattr]
+    fn instances_count() -> u32 {
+        REG_COUNTER.load(std::sync::atomic::Ordering::Relaxed)
+    }
+}
+
+impl PyRegister {
+    /// Correctly performs extraction of size and name during creation of a register
+    pub fn inner_parse_new(
```

Comment:
This method name was confusing at first. It's not parsing anything It's about mapping errors and error messages mostly. I would call it `check_new_inputs()`  or `validate_new_inputs()` to make it clear that this is a validation function not a parser.

Reply:
Gone in 883d82334.